### DISCUSSION TITLE
Feasibility of identical platforms on different boards

### DISCUSSION THREAD

**Discussion by kostareg** [tock/tock#3895] [Q&A] [answered]
I would like to grab an [Aliexpress nRF52832 board](https://www.aliexpress.com/item/4000516424563.html) but before I do I'm curious about how feasible it is to use on the TockOS ecosystem. Since it uses the same platform as the Nordic nRF52-DK, and I will be using a JLink anyways, how much of the process will be plug-and-play (i.e. pretending it is a the DK board) and what kind of troubles can I run into?

P.S. I would like to take advantage of the BLE mesh feature, but it looks like Tock does not support it. I have found [drogue-iot/btmesh](https://github.com/drogue-iot/btmesh) but I'm not sure exactly where to start with that.

**Comment by lschuermann** [comment, marked as answer]
It's definitely feasible to get Tock running on other boards using the nRF52 series of microcontrollers. In fact, the nRF52s are amongst the most widely used and well-tested chips in the Tock ecosystem. For other boards that use these chips, see for instance:

- https://github.com/tock/tock/tree/master/boards/acd52832
- https://github.com/tock/tock/tree/master/boards/clue_nrf52840

If you're using a JLink, porting should be very straightforward. Simply copy the respective `boards/` create and adjust the physical peripherals / pin assignment. The most tricky part is usually supporting foreign board's bootloaders, but that's not required if you have JTAG access.

However, Tock does not have a solid BLE story yet. We can send and receive advertisements, but that's about it. For more information on that, see the various (tracking) issues: #981 #2970
